src/MongoDB.Driver.Core/Core/Authentication/SaslAuthenticator.cs

```diff
@@ -99,13 +95,16 @@ public async Task AuthenticateAsync(IConnection connection, ConnectionDescriptio
         {
             Ensure.IsNotNull(connection, nameof(connection));
             Ensure.IsNotNull(description, nameof(description));
-
             using (var conversation = new SaslConversation(description.ConnectionId))
             {
-                var currentStep = _mechanism.Initialize(connection, conversation, description);
-
```

Comment:
Don't remove all these blank lines.

They serve to separate logically distinct chunks of code.